jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildTarballAndLoadDockerStep.java

```diff
@@ -0,0 +1,131 @@
+    dependencies.addAll(buildApplicationLayerFutures);
+    dependencies.add(NonBlockingFutures.get(buildConfigurationFutureFuture));
+    return Futures.whenAllComplete(dependencies)
+        .call(this::afterPushBaseImageLayerFuturesFuture, listeningExecutorService)
+        .get();
+  }
+
+  /**
+   * Depends on {@code pushAuthorizationFuture}, {@code pushBaseImageLayerFuturesFuture.get()},
+   * {@code pushApplicationLayerFutures}, and (@code
+   * containerConfigurationBlobDescriptorFutureFuture.get()}.
+   *
+   * <p>TODO: Refactor into testable components
+   */
+  private Void afterPushBaseImageLayerFuturesFuture()
+      throws ExecutionException, InterruptedException, IOException {
+    // Add layers to image tarball
+    TarStreamBuilder tarStreamBuilder = new TarStreamBuilder();
+    List<String> layerFiles = new ArrayList<>();
+    for (Future<CachedLayer> cachedLayerFuture :
+        NonBlockingFutures.get(pullBaseImageLayerFuturesFuture)) {
+      Path layerFile = NonBlockingFutures.get(cachedLayerFuture).getContentFile();
+      layerFiles.add(layerFile.getFileName().toString());
+      tarStreamBuilder.addEntry(
+          new TarArchiveEntry(layerFile.toFile(), layerFile.getFileName().toString()));
+    }
+    for (Future<CachedLayer> cachedLayerFuture : buildApplicationLayerFutures) {
+      Path layerFile = NonBlockingFutures.get(cachedLayerFuture).getContentFile();
+      layerFiles.add(layerFile.getFileName().toString());
+      tarStreamBuilder.addEntry(
+          new TarArchiveEntry(layerFile.toFile(), layerFile.getFileName().toString()));
+    }
+
+    // Add config to tarball
+    // TODO: Add ability to add blobs as entries to TarStreamBuilder without using temp files
+    Blob containerConfigurationBlob =
+        NonBlockingFutures.get(NonBlockingFutures.get(buildConfigurationFutureFuture));
+    Path tempConfig = Files.createTempFile(null, null);
+    tempConfig.toFile().deleteOnExit();
+    try (OutputStream bufferedOutputStream = Files.newOutputStream(tempConfig)) {
+      containerConfigurationBlob.writeTo(bufferedOutputStream);
+    }
+    tarStreamBuilder.addEntry(new TarArchiveEntry(tempConfig.toFile(), "config.json"));
+
+    // Add manifest to tarball
+    Blob manifestBlob =
+        DockerLoadManifestBlob.get(buildConfiguration.getTargetImageReference(), layerFiles);
+    Path tempManifest = Files.createTempFile(null, null);
+    tempManifest.toFile().deleteOnExit();
+    try (OutputStream bufferedOutputStream = Files.newOutputStream(tempManifest)) {
+      manifestBlob.writeTo(bufferedOutputStream);
+    }
+    tarStreamBuilder.addEntry(new TarArchiveEntry(tempManifest.toFile(), "manifest.json"));
+
+    // Load the image to docker daemon
+    try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+      tarStreamBuilder.toBlob().writeTo(byteArrayOutputStream);
+      new Command("docker", "load").run(byteArrayOutputStream.toByteArray());
```

Comment:
Although there's already the comment above for refactoring into testable components, this should prob have an explicit comment to not use `Command` since that is an untested and not-very-robust test utility.